src/gc/env/gcenv.ee.h

```diff
@@ -78,7 +80,7 @@ class GCToEEInterface
 
     static void GcEnumAllocContexts(enum_alloc_context_func* fn, void* param);
 
-    static void AttachCurrentThread(); // does not acquire thread store lock
+    static bool CreateBackgroundThread(Thread** thread, GCBackgroundThreadFunction threadStart, void* arg);
```

Comment:
The contract for CreateBackgroundThread should be pretty simple: It should give back a fully initialized background thread to the GC.

The current structure where the background thread initialization is split between EE and GC sides is not good. I think all the steps of the thread initialization - including  `Thread::HasStarted()` and `background_gc_create_event` event - should be done as part of `CreateBackgroundThread`.